File: PersonifiBackend/src/PersonifiBackend.Core/Entities/Transaction.cs
namespace PersonifiBackend.Core.Entities;

public class Transaction
{
    public int Id { get; set; }
    public decimal Amount { get; set; }
    public string Description { get; set; } = string.Empty;
    public string? Notes { get; set; }
    public DateTime TransactionDate { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }

    public int AccountId { get; set; }
    public Account Account { get; set; } = null!;

    public int CategoryId { get; set; }
    public Category Category { get; set; } = null!;

    public int CreatedByUserId { get; set; }
    public User CreatedByUser { get; set; } = null!;
}

File: personifi-app/src/types/transaction.ts
import { CategoryType } from './budget';

export interface Category {
  id: number;
  name: string;
  type: CategoryType;
  icon: string;
  color: string;
}

export interface Transaction {
  id: number;
  amount: number;
  description: string;
  notes?: string;
  transactionDate: string;
  category: Category;
  createdAt: string;
}
